modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -402,6 +415,26 @@ These parameters are used to specify additional options for the encoding process
 */
 CV_EXPORTS_W bool imwriteanimation(const String& filename, const Animation& animation, const std::vector<int>& params = std::vector<int>());
 
+/** @brief Encodes an Animation to a memory buffer.
+
+The function imencodeanimation encodes the provided Animation data into a memory
+buffer in an animated format. Supported formats depend on the implementation and
+may include formats like GIF, AVIF, APNG, or WEBP.
+
+@param ext The file extension that determines the format of the encoded data.
+@param animation A constant reference to an Animation struct containing the
+frames and metadata to be encoded.
+@param buf A reference to a vector of unsigned chars where the encoded data will
+be stored.
+@param params Optional format-specific parameters encoded as pairs (paramId_1,
+paramValue_1, paramId_2, paramValue_2, ...). These parameters are used to
+specify additional options for the encoding process. Refer to `cv::ImwriteFlags`
+for details on possible parameters.
+
+@return Returns true if the animation was successfully encoded; returns false otherwise.
+*/
+CV_EXPORTS_W bool imencodeanimation(const String& ext, const Animation& animation, CV_OUT std::vector<uchar>& buf, const std::vector<int>& params = std::vector<int>());
```

Comment:
I believe `buf` should be of type `OutputArray`

Reply:
All analogs use std::vector:
- imdecode: https://docs.opencv.org/4.x/d4/da8/group__imgcodecs.html#ga4e9883ae1f619bcbe875b7038520ea78
- indecodemulti: https://docs.opencv.org/4.x/d4/da8/group__imgcodecs.html#ga913e55d27f30e7b56f40eddc09fccb76

Most probably, they call std::vector::resize() to fit encoded size. It's not available throw OutputArray interface.